src/cli.ts

```diff
@@ -42,28 +42,38 @@ program
     'API key, to authenticate against the valueset service to be used for resolving missing valuesets',
     undefined
   )
+  .option('-c, --cache-valuesets', 'Whether or not to cache ValueSets retrieved from the ValueSet service', false)
   .parse(process.argv);
 
 function parseBundle(filePath: string): R4.IBundle {
   const contents = fs.readFileSync(filePath, 'utf8');
   return JSON.parse(contents) as R4.IBundle;
 }
 
+function getCachedValueSets(cacheDir: string): R4.IValueSet[] {
+  if (fs.existsSync(cacheDir)) {
+    return fs.readdirSync(cacheDir).map(vs => JSON.parse(fs.readFileSync(path.join(cacheDir, vs), 'utf8')));
+  }
+
+  return [];
+}
+
 async function calc(
   measureBundle: R4.IBundle,
   patientBundles: R4.IBundle[],
-  calcOptions: CalculationOptions
+  calcOptions: CalculationOptions,
+  valueSetCache: R4.IValueSet[] = []
 ): Promise<any> {
```

Comment:
What's the point of defining the return object to be of type <any>? Is it because the different report types differ too greatly for it to make sense to define an interface that encompasses all of them?

Reply:
Short answer: this is laziness on our part.

Usually `any` is really only supposed to be used when either the result of something could really be anything, or the type is not really strictly defined by whatever you're using to get the results. Really, this type should be a concatenation of all the possible return types defined by the functions exported in calculator. As you've pointed out before, a lot of those return types are defined inline in the function declaration. 

I feel like it makes sense to abstract everything to interfaces, and then we could really easily make stricter typing here. Sounds like a good task for people trying to get more familiar with Typescript 👀